pkg/api/handler/tx_finder.go

```diff
@@ -0,0 +1,97 @@
+package handler
+
+import (
+	"context"
+	"encoding/hex"
+	"log/slog"
+
+	"github.com/bitcoin-sv/arc/config"
+	"github.com/bitcoin-sv/arc/internal/validator"
+	"github.com/bitcoin-sv/arc/internal/woc_client"
+	"github.com/bitcoin-sv/arc/pkg/metamorph"
+)
+
+type txFinder struct {
+	th         metamorph.TransactionHandler
+	pc         *config.PeerRpcConfig
+	l          *slog.Logger
+	w          *woc_client.WocClient
+	useMainnet bool
+}
+
+func (f txFinder) GetRawTxs(ctx context.Context, ids []string) ([]validator.RawTx, error) {
+	// NOTE: we can ignore ALL errors from providers, if one returns err we go to another
+
+	// TODO: discuss if it's worth to have implementation for len(ids) == 1
```

Comment:
I don't think it's worth it. I would leave it like that

Reply:
Yeah, you may be right